src/repr/src/scalar.rs

```diff
@@ -3034,8 +3108,8 @@ impl<'a> ScalarType {
             ScalarType::Numeric { .. } => (*NUMERIC).iter(),
             ScalarType::Date => (*DATE).iter(),
             ScalarType::Time => (*TIME).iter(),
-            ScalarType::Timestamp => (*TIMESTAMP).iter(),
-            ScalarType::TimestampTz => (*TIMESTAMPTZ).iter(),
+            ScalarType::Timestamp { .. } => (*TIMESTAMP).iter(),
```

Comment:
These consts should probably get at least one new entry with subseconds, which they currently lack.

Reply:
Ack. This is mostly for testing?

Reply:
Added!

Reply:
Yes, my two comments here help testing. We test all operators and functions with all the types and datums here, so adding the new types and some of their datums gives us much higher confidence this is correct.

Reply:
Makes sense, ty!